include/highfive/H5DataType.hpp

```diff
@@ -136,6 +136,37 @@ class CompoundType : public DataType {
     void create(size_t size = 0);
 };
 
+template <typename T>
+class EnumType : public DataType {
+public:
+    struct member_def {
+        member_def(std::string t_name, T t_value)
+            : name(std::move(t_name))
+            , value(std::move(t_value)) {}
+        std::string name;
+        T value;
+    };
+
+    EnumType(const EnumType& other) = default;
+
+    EnumType(std::vector<member_def> t_members, DataType t_underlying_type)
```

Comment:
I think this constructor doesn't make sense in the context of enum?

Reply:
It's only to be the same than compound type.
Any thoughts @ferdonline ?

Reply:
Maybe I dont see your point @ohm314, to me enums members can have a very similar API than compound, except that there is a single type.